share/eds/byzantine/bad_encoding.go

```diff
@@ -150,13 +150,16 @@ func (p *BadEncodingProof) Validate(header *header.ExtendedHeader) error {
 	if err != nil {
 		return err
 	}
-	rebuiltExtendedShares, err := codec.Encode(rebuiltShares[0 : len(shares)/2])
+
+	squareWidth := uint64(len(header.DAH.RowsRoots) / 2)
```

Comment:
can we still leave this as taking the square width from `merkleRowRoots` ? (since that's len(shares)) -- it's not a diff length and adds a bit of cognitive overhead.

Reply:
sorry about that, forgot I switched while debugging. sure thing! https://github.com/celestiaorg/celestia-node/pull/1338/commits/6986d738d3de5418319f53894b11380e9956dece